Write-Information accept objects from the pipeline

iSazonov opened the pull request:
Close #2559

- [x] Now Write-Information accept objects from the pipeline.
- [x] Add test

At the moment there are no tests for Write-Information (and for Write-Warning too).
@JamesWTruher could you please help?

iSazonov committed:
Write-Information accept objects from the pipeline

daxian-dbw commented:
@iSazonov you can use the PowerShell class to collect the information and warning streams:
```powershell
$ps = [powershell]::Create()
$ps.AddScript("Write-Information 'test' -InformationAction Continue") > $null
$ps.Invoke()
$ps.Streams.Information
```

iSazonov commented:
@daxian-dbw Thanks for comment! I'd use this.

But I have questions remain:
1. Is `Write-Stream.Tests.ps1` a right place for the tests? 
2. Whether enough tests for `Write-Information` and for `Write-Warning`? It seems the number of tests should be much more.

iSazonov committed:
Add test

daxian-dbw commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Stream.Tests.ps1:
```diff
@@ -70,6 +70,24 @@ Describe "Stream writer tests" -Tags "CI" {
+            $rs.open()
+            $ps = [powershell]::Create()
+            $ps.Runspace = $rs
+            $ps = [powershell]::Create()
```
You created 2 PowerShell instances here :)
`$ps = [powershell]::Create()` should be enough here. You don't need to explicitly create the runspace because `$ps.Invoke()` will automatically create a new runspace using `[runspacefactory]::Createrunspace()` is `$ps.Runspace` is null.

daxian-dbw commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Stream.Tests.ps1:
```diff
@@ -70,6 +70,24 @@ Describe "Stream writer tests" -Tags "CI" {
+        AfterAll {
+            $rs.Close()
+            $rs.Dispose()
+            $ps.Dispose()
```
So here you only need `$ps.Dispose()`.

daxian-dbw commented:
> Is Write-Stream.Tests.ps1 a right place for the tests? 

Yes, that looks like the right place.

> Whether enough tests for Write-Information and for Write-Warning? It seems the number of tests should be much more.

I'm not sure how many tests we have for the stream cmdlets. Maybe @JamesWTruher has better idea.

iSazonov commented:
@daxian-dbw Thanks for clarifications!
I think open new issue that we need more tests.

iSazonov committed:
Fix test

Remove unneeded runspace.

daxian-dbw committed:
Clear streams before each test case

Streams of the PowerShell instance might have records from other test runs. Clear the streams before each test case.

daxian-dbw commented:
Thank you @iSazonov for the fix!

iSazonov commented:
@daxian-dbw Thanks for review! Your comments are always very valuable!
